pysaliency/dataset_config.py

```diff
@@ -52,6 +54,115 @@ def apply_dataset_filter_config(stimuli, fixations, filter_config):
     return filter_fn(stimuli, fixations, **filter_config['parameters'])
 
 
+def filter_scanpaths_by_attribute(scanpaths: FixationTrains, whitelist: dict=None, blacklist: dict=None):
+    """Filter Scanpaths by values of scanpath attributes (fixation_trains.scanpath_attributes), the dictionary can have only one attribute"""
+    
+    assert (whitelist is None and blacklist is not None) or (whitelist is not None and blacklist is None)
+    if whitelist is not None:
+        assert(len(whitelist)==1)
```

Comment:
my suggestion last friday was to only implement for now functions
```
def filter_scanpaths_by_attribute(scanpaths, attribute_name, attribute_value, invert_match=False):
    ....
```

They  would be sufficiently convenient for many applications and are much easier to write. What you already started to implement would be `filter_scanpaths_by_attributes` (note the trailing `s`), which then could simply call `filter_scanpaths_by_attribute` repeated times, or do a more efficient filter. But for now let's only for for the simple version that I just outlined. It will be easier to read and to use

Reply:
`invert_match` is intended to choose between filtering positives or negatives